### ISSUE TITLE
SFML_ROOT on iOS does not work properly

### ISSUE THREAD

**Issue by Zombieschannel** [SFML/SFML#3360]
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

SFML 3.0.0 with CMake's `find_package(SFML 3.0 ...)` seems to also require manually adding `find_package(Ogg)`, otherwise there is this error message.
```
By not providing "FindOgg.cmake" in CMAKE_MODULE_PATH this project has
asked CMake to find a package configuration file provided by "Ogg", but
CMake did not find one.
Could not find a package configuration file provided by "Ogg" with any of
the following names:
OggConfig.cmake
ogg-config.cmake
Add the installation prefix of "Ogg" to CMAKE_PREFIX_PATH or set "Ogg_DIR"
to a directory containing one of the above files.  If "Ogg" provides a
separate development package or SDK, be sure it has been installed.
```

I assume this is a bug since it is not mentioned anywhere in the migration guide nor does it happen with any other dependency.

### Your Environment

- OS / distro / window manager: Windows
- SFML version: 3.0.0
- Compiler / toolchain: MSVC

### Steps to reproduce

Use CMake's find package on SFML 3

### Expected behavior

-

### Actual behavior

-

**Comment by eXpl0it3r** [comment]
Where did you get SFML from, how do you tell CMake where to find SFML?

**Comment by Zombieschannel** [comment]
Yeah ok setting SFML_ROOT instead of SFML_DIR seems to have fixed this.
Though I thought that SFML_DIR was the way to go..? https://en.sfml-dev.org/forums/index.php?topic=24070.0

**Comment by eXpl0it3r** [comment]
Looks like we forgot to add that information to the migration guide even though we talked about it before the release 😅

> Though I thought that SFML_DIR was the way to go..? https://en.sfml-dev.org/forums/index.php?topic=24070.0

I thought that as well, but either the stance around that changed, or we were misinformed from the start 🙃

If you install libs to the same place, you can actually use CMAKE_PREFIX_PATH just once for all of them.

See also:
- https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure
- https://cmake.org/cmake/help/latest/command/find_path.html

**Comment by ChrisThrasher** [comment]
> Yeah ok setting SFML_ROOT instead of SFML_DIR seems to have fixed this. Though I thought that SFML_DIR was the way to go..? https://en.sfml-dev.org/forums/index.php?topic=24070.0

If you look at the `find_package` docs you’ll see that `<PackageName>_ROOT` is the highest priority variable to determine where to find a package so we chose to stop using or recommending `SFML_ROOT`.

Apologies for not making this clear in migration guide. It may be worth adding a new section to explain the switch.

**Comment by Zombieschannel** [comment]
Hm, on iOS SFML_ROOT does not seem to work.

![image](https://github.com/user-attachments/assets/af9c4078-3a20-46b6-97ae-10e069dc9380)

I've tried changing the path to be closer to what SFML_DIR is but no luck.

Can anyone else confirm having the same issue?

This is fixable with the following lines before `find_package` but it does use those _DIR variables.
```
    set(Freetype_DIR ${SFML_ROOT}/lib/cmake/freetype)
    set(Vorbis_DIR ${SFML_ROOT}/lib/cmake/Vorbis)
    set(Ogg_DIR ${SFML_ROOT}/lib/cmake/Ogg)
    set(FLAC_DIR ${SFML_ROOT}/lib/cmake/FLAC)
    set(SFML_DIR ${SFML_ROOT}/lib/cmake/SFML)
```

**Comment by JonnyPtn** [comment]
using `SFML_ROOT` Is a bit tricky with iOS, as by default it will only search for packages in the target sys root (i.e. the iOS sdk folder). You can override this by setting `CMAKE_FIND_ROOT_PATH_MODE_PACKAGE` to `BOTH` so it also searches on the host system. This is possibly a risky thing to do, as it means cmake may find incorrect libraries (e.g. it might find the Mac build of SFML or its dependencies). I don't think there's anything for SFML to do about that, it's just a cmake quirk

`SFML_DIR` on the other hand is still usable and **not** something we should be avoiding (as you can see, cmake output even suggests it) - I see the same issue as you originally reported and expect it's similar to above in that SFML's config is not searching outside the iOS sdk for it's dependencies so doesn't find them, which should be fixable
